ebs_snapshots/daemon.py

```diff
@@ -0,0 +1,209 @@
+'''
+***
+Modified generic daemon class
+***
+
+Source:         https://github.com/serverdensity/python-daemon
+
+Author:         http://www.jejik.com/articles/2007/02/
+                        a_simple_unix_linux_daemon_in_python/www.boxedice.com
+
+License:        http://creativecommons.org/licenses/by-sa/3.0/
+'''
+
+# Core modules
+import atexit
+import os
+import sys
+import time
+import signal
+
+
+class Daemon(object):
+    """
+    A generic daemon class.
+
+    Usage: subclass the Daemon class and override the run() method
+    """
+    def __init__(self, pidfile, stdin=os.devnull,
+                 stdout=os.devnull, stderr=os.devnull,
+                 home_dir='.', umask=022, verbose=1):
```

Comment:
This is taken from public daemon implementation - would rather follow it in terms of conventions like this https://github.com/serverdensity/python-daemon